sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -82,6 +87,8 @@ object NestedColumnAliasing {
     case _: LocalLimit => true
     case _: Repartition => true
     case _: Sample => true
+    case _: RepartitionByExpression => true
+    case _: Join => true
```

Comment:
Btw, joins update the nullability of input attributes (outer join cases), but does this logic work correctly? Either way, I think we need some tests for the case. It seems the current PR only has tests for inner join cases?

Reply:
Yea, I will add more tests for join, like outer join.

The nested column pruning here is for a Project on top of Join (or others). We just replace original complex output with nested pruned fields. IIUC, although joins update nullability, for null outputs, those nested pruned fields should be null too, because the nullability is determined by join semantics.

I will add more tests to verify it.

Reply:
Added some tests for outer join.